### PR TITLE
Use window leaflet

### PR DESCRIPTION
**Pull request by Jesus89**
Related to: https://github.com/CartoDB/cartodb.js/issues/1974

This PR fixes:
- Leaflet was required before Carto.js in the browser
- Leaflet was required using Carto.js in node

### CODE REVIEW

**Comment by Jesus89** [comment]
I think we can assume that it comes from window, as the `google` lib.

With the `/* global X */` tags in the beginning of the file we can see that it comes from the global scope.

**Comment by IagoLast** [comment]
Lets vote!

@alonsogarciapablo @ivanmalagon

File: src/cartodb.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
-var isLeafletAlreadyLoaded = !!window.L;
+window.L = require('leaflet');
```
Comment (alonsogarciapablo) [inline src/cartodb.js]: Shouldn't we remove this too?

**Comment by alonsogarciapablo** [comment]
I'm fine with using `L` directly. We all know where it comes from.

File: src/cartodb.js (hunk 1 above)
Comment (Jesus89) [reply src/cartodb.js]: This is the only require used. It is used by the internal bundle if we remove this line the Builder do not include Leaflet in the window context.

**Pull request merged**